lib/net/pool.js

```diff
@@ -1912,6 +1908,9 @@ class Pool extends EventEmitter {
     const blocks = [];
 
     while (hash) {
+      if (!await this.chain.getBlock(hash))
+        break;
+
```

Comment:
I'm not sure this is the right logic here. Also there's still a `if(prune)` up on line 1900.

I think what we may need to do is, after `hash = findLocator`, check the height of that block hash and if we are pruned and that height is within 288 of tip then continue, otherwise abort (`return`)